### PR TITLE
Stabilize tests: fix sysfs error, select only supported devices, add a reporter

### PR DESCRIPTION
**Pull request by fedepaol**
Fix sysfs error in case of deeper pci addresses.
Select only officially supported devices.
Add a reporter that dumps the state of various CRs.

### CODE REVIEW

**Comment by fedepaol** [comment]
/hold
still need to validate this

File: pkg/k8sreporter/reporter.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+	r.logSriovNodeState()
+	r.logNetworkPolicies()
+}
+
+// Cleanup cleans up the current content of the artifactsDir
+func (r *KubernetesReporter) Cleanup() {
+}
+
+func (r *KubernetesReporter) logPods(namespace string) {
+	pods, err := r.clients.Pods(v1.NamespaceAll).List(metav1.ListOptions{})
```
Comment (SchSeba) [inline pkg/k8sreporter/reporter.go]: Can you use the namespace here?

File: pkg/k8sreporter/reporter.go [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+	j, err := json.MarshalIndent(pods, "", "    ")
+	if err != nil {
+		fmt.Println("Failed to marshal pods", err)
+		return
+	}
+	fmt.Fprintln(r.dumpOutput, string(j))
+}
+
+func (r *KubernetesReporter) logResource(resourceName string) {
```
Comment (SchSeba) [inline pkg/k8sreporter/reporter.go]: This function is not in use

File: pkg/k8sreporter/reporter.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+		fmt.Fprintf(os.Stderr, "failed to fetch pods: %v\n", err)
+		return
+	}
+
+	for _, pod := range pods.Items {
+		if filterPods(&pod) {
+			continue
+		}
+		for _, container := range pod.Spec.Containers {
+
```
Comment (SchSeba) [inline pkg/k8sreporter/reporter.go]: nit: can you remove this empty line here :)

File: pkg/k8sreporter/reporter.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	j, err := json.MarshalIndent(networks, "", "    ")
+	if err != nil {
+		fmt.Println("Failed to marshal networks")
+		return
+	}
+	fmt.Fprintln(r.dumpOutput, string(j))
+}
+
+func (r *KubernetesReporter) logSriovNodeState() {
+	nodeStates, err := r.clients.SriovNetworkNodeStates(v1.NamespaceAll).List(metav1.ListOptions{})
```
Comment (SchSeba) [inline pkg/k8sreporter/reporter.go]: Can we use the operator namespace here also?

**Review by SchSeba — CHANGES_REQUESTED** [review]
Just a few comments and it should be ready to go

File: pkg/k8sreporter/reporter.go (hunk 4 above)
Comment (fedepaol) [reply pkg/k8sreporter/reporter.go]: That should not make a difference, but ok

**Review by SchSeba — APPROVED** [review]
/lgtm

**Comment by SchSeba** [comment]
/hold

wait until validation

**Pull request merged**